core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionProvider.java

```diff
@@ -57,4 +57,37 @@ default FileObject getDirectFile( T connectionDetails, String path ) throws Kett
     return KettleVFS.getFileObject( pvfsUrl, new Variables(), getOpts( connectionDetails ) );
   }
 
+
+  /**
+   * Test the VFS connection with provided connection details .
+   *
+   *
+   * @return {@code true} if the provided rootPath is valid
+   */
+  default boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
+    return test( connectionDetails );
+  }
+
+  /**
+   * Method to resolve root path for a given VFS connection.
+   *
+   *
+   * @return resolved root path.
+   */
+
+  default String getResolvedRootPath( T connectionDetails ) {
+    return connectionDetails.getRootPath();
+  }
+
+  /**
+   * Method check if a VFS connection has buckets.
```

Comment:
```java
/**
 * Checks if a given connection uses buckets in its current configuration.
 * <p>
 * A connection is using buckets if it can have buckets, as determined by {@link ConnectionDetails#hasBuckets()}, and if its {@link #getResolvedRootPath() resolved root path} is empty.
 *